### PR TITLE
F#54 add lidar to robot's urdf and simulation

### PR DESCRIPTION
**Pull request by LyubomyrD**
Added Hokuyo LIDAR sensor

Closing #54

### CODE REVIEW

File: leobot_description/CMakeLists.txt [hunk 1]
```diff
@@ -3,13 +3,16 @@ project(leobot_description)
 
 find_package(catkin REQUIRED)
 
-catkin_package()
+catkin_package(
+  CATKIN_DEPENDS urg_node
```
Comment (AndriyPt) [inline leobot_description/CMakeLists.txt]: For simulation you do not need *urg_node*.

File: leobot_description/package.xml [hunk 2]
```diff
@@ -7,4 +7,5 @@
   <license>MIT</license>
 
   <buildtool_depend>catkin</buildtool_depend>
+  <run_depend>urg_node</run_depend>
```
Comment (AndriyPt) [inline leobot_description/package.xml]: For simulation you do not need *urg_node*.

File: leobot_description/urdf/sensors/hokuyo.urdf.xacro [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
+     <xacro:property name="hokuyo_width" value="0.001"/>
+     <xacro:property name="hokuyo_height" value="0.05"/>
+     <xacro:property name="hokuyo_depth" value="0.06"/>
+
```
Comment (AndriyPt) [inline leobot_description/urdf/sensors/hokuyo.urdf.xacro]: please remove extra blank line

File: leobot_description/urdf/sensors/hokuyo.urdf.xacro [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
```
Comment (AndriyPt) [inline leobot_description/urdf/sensors/hokuyo.urdf.xacro]: Please fix formatting of XML file

File: leobot_description/urdf/sensors/hokuyo.urdf.xacro [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+     <xacro:property name="hokuyo_width" value="0.001"/>
+     <xacro:property name="hokuyo_height" value="0.05"/>
+     <xacro:property name="hokuyo_depth" value="0.06"/>
+
+
+  <!-- Hokuyo 2D LIDAR -->
```
Comment (AndriyPt) [inline leobot_description/urdf/sensors/hokuyo.urdf.xacro]: Name of macros is stating that :)

File: leobot_description/urdf/sensors/hokuyo.urdf.xacro [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  <xacro:macro name="LIDAR_hokuyo" params="base_radius base_height">
```
Comment (AndriyPt) [inline leobot_description/urdf/sensors/hokuyo.urdf.xacro]: please use lower case for macros names

**Review by AndriyPt — CHANGES_REQUESTED** [review]

**Comment by LyubomyrD** [comment]
@AndriyPt now LIDAR works as expected. Please, have another look at the corrections that You requested. I tried to do my best to fix them.

File: leobot_description/package.xml [hunk 7]
```diff
@@ -7,4 +7,6 @@
   <license>MIT</license>
 
   <buildtool_depend>catkin</buildtool_depend>
+  <run_depend>sensor_msgs</run_depend>
+  <build_depend>urg_node</build_depend>
```
Comment (AndriyPt) [inline leobot_description/package.xml]: This node is not needed for simulation. Please remove it.

**Review by AndriyPt — CHANGES_REQUESTED** [review]
Small change and happy to merge

File: leobot_description/package.xml (hunk 7 above)
Comment (LyubomyrD) [reply leobot_description/package.xml]: @AndriyPt sensor_msg is removed

**Comment by LyubomyrD** [comment]
@AndriyPt sensor_msg - removed

File: leobot_description/package.xml (hunk 7 above)
Comment (AndriyPt) [reply leobot_description/package.xml]: urg_node as well :)

**Review by AndriyPt — CHANGES_REQUESTED** [review]

File: leobot_description/robot/leobot.gazebo.xacro [hunk 8]
```diff
@@ -127,4 +127,38 @@
     <material>Gazebo/Green</material>
   </gazebo>
 
+  <!-- Hokuyo LIDAR for simulation -->
+   <gazebo reference="laser_link">
+      <material>Gazebo/Green</material>
+      <sensor type="ray" name="laser">
+        <pose>0 0 0 0 0 0</pose>
+        <visualize>true</visualize>
```
Comment (AndriyPt) [inline leobot_description/robot/leobot.gazebo.xacro]: Please hide LIDAR visualization from Simulation. It is better seen in the RViz no need to overload simulator :)

**Review by AndriyPt — CHANGES_REQUESTED** [review]

**Review by AndriyPt — APPROVED** [review]

**Pull request merged**